### PR TITLE
Add power

### PR DESCRIPTION
**Pull request by gujor**
This PR adds the `Power` characteristics and enables these device properties:
- ACTIVE_POWER_LOAD_SIDE
- LUMINAIRE_NOMINAL_INPUT_POWER
- LUMINAIRE_POWER_AT_MINIMUM_DIM_LEVEL
- PRESENT_DEVICE_INPUT_POWER

power
Part of a screenshot from the example app showing `Power` usage.

### CODE REVIEW

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/DeviceProperty.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package no.nordicsemi.android.mesh.sensorutils;
 
 import androidx.annotation.NonNull;
+
```
Comment (jjenequi) [inline mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/DeviceProperty.java:4]: Empty line?

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * The Power characteristic is used to represent a measure of power in units of watts.
+ */
+public class Power extends DevicePropertyCharacteristic<Float> {
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY)
+    public Power(@NonNull final byte[] data, final int offset, final int length) {
+        super(data, offset, length);
+        int bits = 0;
+        byte[] bytes = Arrays.copyOfRange(data, offset, offset + length);
+        for (byte b : bytes) bits = (bits >> 8) | (((int) b & 0xFF) << 16);
```
Comment (jjenequi) [inline mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java:18]: Shouldn't this be << 24?

Comment (jjenequi) [reply mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java:18]: Also, aren't there any standard library way of doing these kind of operations?

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public int getLength() {
+        return 3;
+    }
+
+    @Override
+    public byte[] getBytes() {
+        byte[] bytes = {(byte) 0xFF, (byte) 0xFF, (byte) 0xFF};
+        if (this.value != null) {
+            int value = (int) (this.value * 10.0f + 0.5d);
```
Comment (jjenequi) [inline mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java:47]: Why mixing types 10.0 (float) and 0.5 (double)?

File: mesh/src/test/java/no/nordicsemi/android/mesh/sensorutils/DevicePropertyCharacteristicTest.java [hunk 4]
```diff
@@ -180,4 +180,39 @@ public void testDateUTC() {
...
+        final ArrayList<Float> expectedSamples = new ArrayList<>();
+        expectedSamples.add(0.0f);
+        expectedSamples.add(0.1f);
+        expectedSamples.add(838860.7f);
+        expectedSamples.add(1677721.4f);
+        expectedSamples.add(null);
+        final ArrayList<byte[]> samples = new ArrayList<>();
+        samples.add(new byte[]{0x00, 0x00, 0x00});
+        samples.add(new byte[]{0x01, 0x00, 0x00});
+        samples.add(new byte[]{(byte) 0xFF, (byte) 0xFF, 0x7F});
```
Comment (jjenequi) [inline mesh/src/test/java/no/nordicsemi/android/mesh/sensorutils/DevicePropertyCharacteristicTest.java:195]: Perhaps the test should include different values for first and second bytes to catch the bug in the code?

**Review by jjenequi — CHANGES_REQUESTED** [review]

**Review by jjenequi — COMMENTED** [review]
From readme in the real app, this commit shouldn't be needed when building.

### Step 2: Create dummy publishing values in local.settings
The mesh SDK includes publishing tasks that require some properties to be set. Your project should already include
a 'mesh/local.properties' file, in which you need to add the following:

```
sonatypeStagingProfileId=id
ossrhUsername=user
ossrhPassword=pwd

signing.keyId=id
signing.password=pwd
signing.secretKeyRingFile=file
```
If there is no such file, you can safely create it empty and add the text as well. Make sure you do not add
to the 'local.properties' that is located in the project root, it must be in the mesh subdirectory.

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/DeviceProperty.java (hunk 1 above)
Comment (gujor) [reply mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/DeviceProperty.java]: Android Studio default formatting inserts empty line, I have now removed it to not confuse :)

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java (hunk 2 above)
Comment (gujor) [reply mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java]: I think shifting with 16 is correct since it is a 24 bit number. But I have changed it to a non-loop version now. I don't know about any standard library ways of dealing with this, especially since it's a 24 bit number. But I don't know Java so ...

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java (hunk 3 above)
Comment (gujor) [reply mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Power.java]: Fixed copy&paste error

File: mesh/src/test/java/no/nordicsemi/android/mesh/sensorutils/DevicePropertyCharacteristicTest.java (hunk 4 above)
Comment (gujor) [reply mesh/src/test/java/no/nordicsemi/android/mesh/sensorutils/DevicePropertyCharacteristicTest.java]: I have added the suggested case and also padding beginning and end with bytes to test offset.

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/DeviceProperty.java [hunk 5]
```diff
@@ -956,9 +956,9 @@ public static String getPropertyName(final DeviceProperty deviceProperty) {
     }
 
     public static DevicePropertyCharacteristic<?> getCharacteristic(@NonNull final DeviceProperty deviceProperty,
-                                                                    @NonNull final byte[] data,
-                                                                    int offset,
-                                                                    final int length) {
+            @NonNull final byte[] data,
```
Comment (jjenequi) [inline mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/DeviceProperty.java:959]: A bit picky but this shouldn't really be part of this commit, or?

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/DeviceProperty.java [hunk 6]
```diff
@@ -982,7 +982,7 @@ public static String getPropertyName(final DeviceProperty deviceProperty) {
             case PRESENT_DEVICE_OPERATING_TEMPERATURE:
                 return new Temperature(data, offset, 2);
             case PEOPLE_COUNT:
-                return new Count(data, offset, 2); //Count16
+                return new Count(data, offset, 2); // Count16
```
Comment (jjenequi) [inline mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/DeviceProperty.java:985]: Same for this one.

File: mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Energy32.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * The Energy 32 characteristic is used to represent a measure of energy in units of kilowatt-hours.
+ */
+public class Energy32 extends DevicePropertyCharacteristic<Double> {
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY)
+    public Energy32(@NonNull final byte[] data, final int offset, final int length) {
+        super(data, offset, length);
+        long bits = 0;
+        byte[] bytes = Arrays.copyOfRange(data, offset, offset + length);
```
Comment (jjenequi) [inline mesh/src/main/java/no/nordicsemi/android/mesh/sensorutils/Energy32.java:17]: Cool, this lovely for loop and copy of range :)
